python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -0,0 +1,252 @@
+                f" The provided endpoint is for a {model_info.model_type} model."
+            )
+
+        return (
+            ChatCompletionsClientAsync(
+                endpoint=azure_ai_inference_settings.endpoint,
+                credential=AzureKeyCredential(azure_ai_inference_settings.api_key.get_secret_value()),
+            ),
+            model_info,
+        )
+
+    def _get_metadata_from_response(self, response: ChatCompletions | AsyncStreamingChatCompletions) -> dict[str, Any]:
+        """Get metadata from the response.
+
+        Args:
+            response: The response from the service.
+
+        Returns:
+            A dictionary containing metadata.
+        """
+        return {
+            "id": response.id,
+            "model": response.model,
+            "created": response.created,
+            "usage": response.usage,
+        }
+
+    def _create_chat_message_content(
+        self, response: ChatCompletions, choice: ChatChoice, metadata: dict[str, Any]
+    ) -> ChatMessageContent:
+        """Create a chat message content object.
+
+        Args:
+            response: The response from the service.
+            choice: The choice from the response.
+            metadata: The metadata from the response.
+
+        Returns:
+            A chat message content object.
+        """
+        return ChatMessageContent(
+            role=AuthorRole(choice.message.role),
+            content=choice.message.content,
+            inner_content=response,
+            finish_reason=FinishReason(choice.finish_reason),
+            metadata=metadata,
+        )
+
+    def _create_streaming_chat_message_content(
+        self,
+        chunk: AsyncStreamingChatCompletions,
+        choice: StreamingChatChoiceUpdate,
+        metadata: dict[str, Any],
+    ) -> StreamingChatMessageContent:
+        """Create a streaming chat message content object.
+
+        Args:
+            chunk: The chunk from the response.
+            choice: The choice from the response.
+            metadata: The metadata from the response.
+
+        Returns:
+            A streaming chat message content object.
+        """
+        return StreamingChatMessageContent(
+            role=AuthorRole(choice.delta.role),
+            content=choice.delta.content,
+            choice_index=choice.index,
+            inner_content=chunk,
+            finish_reason=FinishReason(choice.finish_reason),
+            metadata=metadata,
+        )
+
+    def _format_chat_history(self, chat_history: ChatHistory) -> list[ChatRequestMessage]:
+        """Format the chat history to the expected objects for the client.
+
+        Args:
+            chat_history: The chat history.
+
+        Returns:
+            A list of formatted chat history.
+        """
+        self._prepare_chat_history_for_request(chat_history)
+        return [MessageConverter[message.role](content=message.content) for message in chat_history.messages]
```

Comment:
message.content can only contain text, this should work with the `.items` to allow multi-modal content...